flow/layers/display_list_raster_cache_item.cc

```diff
@@ -0,0 +1,176 @@
+      !context->raster_cached_entries) {
+    return;
+  }
+  auto* raster_cache = context->raster_cache;
+  SkRect bounds = display_list_->bounds().makeOffset(offset_.x(), offset_.y());
+  // We must to create an entry whenever if the react is intersect.
+  // if the rect is intersect we will get the entry access_count to confirm if
+  // it great than the threshold. Otherwise we only increase the entry
+  // access_count.
+  if (context->cull_rect.intersect(bounds)) {
+    if (raster_cache->MarkSeen(key_id_, transformation_matrix_) <
+        raster_cache->access_threshold()) {
+      cache_state_ = CacheState::kNone;
+      return;
+    }
+    context->subtree_can_inherit_opacity = true;
+    cache_state_ = CacheState::kCurrent;
+  } else {
+    raster_cache->Touch(key_id_, matrix);
+  }
+  return;
+}
+
+bool DisplayListRasterCacheItem::Draw(const PaintContext& context,
+                                      const SkPaint* paint) const {
+  return Draw(context, context.leaf_nodes_canvas, paint);
+}
+
+bool DisplayListRasterCacheItem::Draw(const PaintContext& context,
+                                      SkCanvas* canvas,
+                                      const SkPaint* paint) const {
+  if (!context.raster_cache || !canvas) {
+    return false;
+  }
+  if (cache_state_ == CacheState::kCurrent) {
+    return context.raster_cache->Draw(key_id_, *canvas, paint);
+  }
+  // This display_list doesn't cache itself, this only increase the entry
+  // access_count;
+  context.raster_cache->Touch(key_id_, canvas->getTotalMatrix());
+  return false;
+}
+
+static const auto* flow_type = "RasterCacheFlow::DisplayList";
+
+bool DisplayListRasterCacheItem::TryToPrepareRasterCache(
+    const PaintContext& context,
+    bool parent_cached) const {
+  // If we don't have raster_cache we should not cache the current display_list.
+  // If the current node's ancestor has been cached we also should not cache the
+  // current node. In the current frame, the raster_cache will collect all
+  // display_list or picture_list to calculate the memory they used, we
+  // shouldn't cache the current node if the memory is more significant than the
+  // limit.
+  if (cache_state_ == kNone || !context.raster_cache || parent_cached ||
+      !context.raster_cache->GenerateNewCacheInThisFrame()) {
+    return false;
+  }
+  SkRect bounds = display_list_->bounds().makeOffset(offset_.x(), offset_.y());
+  RasterCache::Context r_context = {
+      // clang-format off
+      .gr_context         = context.gr_context,
+      .dst_color_space    = context.dst_color_space,
+      .matrix             = transformation_matrix_,
+      .logical_rect       = bounds,
+      .flow_type          = flow_type,
+      .checkerboard       = context.checkerboard_offscreen_layers,
+      // clang-format on
+  };
+  return context.raster_cache->UpdateCacheEntry(
+      GetId().value(), r_context,
+      [=](SkCanvas* canvas) { display_list_->RenderTo(canvas); });
```

Comment:
Please explicitly capture only what you need, e.g. `[display_list = display_list_, canvas = cavnas]`